src/layout/ascii_layout_spec.cpp

```diff
@@ -187,12 +193,30 @@ void LayoutFromSpec::init_cache(const AsciiLayoutSpec& spec)
 
     }
 
+    if (magic_states_reserved_) {
+        int reserved_count = 0;
+        for (unsigned int i = 0; i < cached_distilled_state_locations_.size(); i++) {
+            for (unsigned int j = 0; j< cached_distilled_state_locations_[i].size(); j++) {
+                reserved_count++;
+            }
+        }
+        try {
+            if (reserved_count != reserved_for_magic_state_cells.size()) {
+                throw std::string("Cells reserved for magic states must be adjacent to distillation regions. Converting mis-used 'M' to 'r'.");
+            }
+        }
+        catch (std::string& error) {
+            std::cerr << error << std::endl;
```

Comment:
No need to throw an exception here

Reply:
I would prefer a this to be a failure. So, just throwing an `std::runtime_error` with no try-catch.

The reason for this is that handling invalid situations can easily result in hard to debug situations in the long run